### DISCUSSION TITLE
Hide JSON escape char in explore logs when using "Show this field"

### DISCUSSION THREAD

**Discussion by Mario-Hofstaetter** [grafana/grafana#48693] [Dashboards, Panels & Visualizations]
TL;DR: When using ` | json | line_format "{{.Message}}"` , JSON escape quotes like `\"Foobar\"` are removed and only shown as `"Foobar"` ✔
When using "Show this field instead of the message" in explore, `\"` backslashes are visible.

Can the backslash be hidden in explore if  `| json` is used in the query?

----

We are logging JSON logs to loki that use message templating for structed logs.
We use [serilog](https://github.com/serilog/serilog/wiki/Structured-Data) and a serilog loki sink for that.

This results in a JSON line (shortened) like this:

```json
{
    "Timestamp": "2022-05-04T12:59:34.6097617+02:00",
    "Level": "Information",
    "MessageTemplate": "Starting application {ApplicationName}:{ApplicationVersion} in environment {HostEnvironment}",
    "Properties": {
        "ApplicationName": "AX.Service.WebBuilder",
        "ApplicationVersion": "2.0.53.0",
        "HostEnvironment": "Development",
    }
}
```
The [serilog-sinks-grafana-loki](https://github.com/serilog-contrib/serilog-sinks-grafana-loki) package is also adding a rendered `Message` Property, **where String Properties are automatically surrounded by double quotes `"Foobar"`**

Viewing this in grafana explore with loki datasource (LogQL uses `| json` transformation), it looks like this, notice the `\"` escapes:

![grafik](https://user-images.githubusercontent.com/33002073/166671889-a03731d3-e83a-4054-b879-f0e860c58e14.png)

After adding: `| line_format "{{.Message}}"` it looks **perfect** (quotes without backslash):

![grafik](https://user-images.githubusercontent.com/33002073/166672218-4294d4b9-5ba7-4652-9aeb-3bddf492d56b.png)

When using "Show this field instead of the message" with `Message`, logs are displayed like this:

![grafik](https://user-images.githubusercontent.com/33002073/166672601-d41fc4f3-367d-4118-887a-d60db9343b2f.png)

The log properties are displayed like this:

![grafik](https://user-images.githubusercontent.com/33002073/166672353-ce6a0895-7c95-4e15-b608-1d3752245c9d.png)

Is it possible to get rid of the backslashes in front of quotes?

(Grafana 8.5.0, Loki 2.5.0)

**Comment by bruno-notifi** [comment]
same. +1

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
